include/picongpu/plugins/saxs/Saxs.hpp

```diff
@@ -0,0 +1,397 @@
+            sumfcoskr = new GridBuffer<float1_64, DIM1 > (DataSpace<DIM1> (n_q));
+            sumfsinkr = new GridBuffer<float1_64, DIM1 > (DataSpace<DIM1> (n_q));
+            np = new GridBuffer<float1_X, DIM1 > (DataSpace<DIM1> (1)); //create one float on GPU and host
+            nmp = new GridBuffer<int64_t, DIM1 > (DataSpace<DIM1> (1)); //create one int on GPU and host
+            sumfcoskr_master = new float1_64[n_q];
+            sumfsinkr_master = new float1_64[n_q];
+            intensity_master = new float1_64[n_q];
+            Environment<>::get().PluginConnector().setNotificationPeriod(this, notifyPeriod);
+            pmacc::Filesystem<simDim>& fs = Environment<simDim>::get().Filesystem();
+
+            if(isMaster)
+            {
+                fs.createDirectory("saxsOutput");
+                fs.setDirectoryPermissions("saxsOutput");
+            }
+        }
+    }
+
+    void writeQintensity(float1_64* intensity, std::string name)
+    {
+        std::ofstream ofile;
+        ofile.open(name.c_str(), std::ofstream::out | std::ostream::trunc);
+        if(!ofile)
+        {
+            std::cerr << "Can't open file [" << name << "] for output, disable plugin output.\n";
+            isMaster = false; // no Master anymore -> no process is able to write
+        }
+        else
+        {
+            int i_x, i_y, i_z;
+            float3_64 q;
+
+            ofile << n_q << "\n";
+            ofile << "# qx qy qz intensity \n";
+            for (unsigned int i = 0; i < n_q ; i++)
+            {
+                i_z = i % n_qz ;
+                i_y = (i / n_qz) % n_qy ;
+                i_x = i / (n_qz * n_qy) ;
+                q[0] = q_min[0] + q_step[0] * i_x;
+                q[1] = q_min[1] + q_step[1] * i_y;
+                q[2] = q_min[2] + q_step[2] * i_z;
+                ofile << q[0] << " " << q[1] << " " << q[2] << " " << intensity[i][0] << "\n";
+            }
+
+            ofile.close();
+        }
+    }
+
+    void writeLog(float1_X np_master,int64_t nmp_master, std::string name)
+    {
+        std::ofstream ofile;
+        ofile.open(name.c_str(), std::ofstream::out | std::ostream::trunc);
+        if(!ofile)
+        {
+            std::cerr << "Can't open file [" << name << "] for output, disable plugin output.\n";
+            isMaster = false; // no Master anymore -> no process is able to write
+        }
+        else
+        {
+            ofile <<  "Number of particles:"  << " " << np_master.x() << "\n";
+            ofile <<  "Number of macro particles:"  << " " << nmp_master << "\n";
+
+            ofile.close();
+        }
+    }
+
+
+    void pluginUnload()
+    {
+        if(!notifyPeriod.empty())
+        {
+            __delete(sumfcoskr);
+            __delete(sumfsinkr);
+            __delete(np);
+            __delete(nmp);
+            __deleteArray(intensity_master);
+            CUDA_CHECK(cudaGetLastError());
+        }
+    }
+
+    /** Method to copy data from GPU to CPU */
```

Comment:
please change to single line comment (`// Method ...`) or add more description to this method :)